components/ProposalRow.tsx

```diff
@@ -1,146 +1,337 @@
 import clsx from 'clsx';
-import { truncate } from '@/util';
+import { nFormatter } from '@/util';
 import PillAction from '@/components/PillAction';
 import { formatUnits } from 'viem';
 import { Proposal } from '@/hooks/useProposals';
 import { useConfig } from '@/hooks/useConfig';
 import { useEasyWrite } from '@/hooks/useEasyWrite';
 import { parseAbi } from 'viem';
 import { useFees } from '@/hooks/useFees';
+import Image from 'next/image';
 import { useWalletClient } from 'wagmi';
 export default function ProposalRow({
   id,
   isBridged,
   description,
   startBlock,
+  status,
   votes,
   tallyLink,
   votingPower,
 }: Proposal) {
-  const { l1, l2 } = useConfig();
+  const { l1, l2, daoLogo } = useConfig();
   const { fees } = useFees();
-  const { data: walletClient } = useWalletClient();
-  const {
-    write: bridgeProposal,
-    isLoading: bridgeProposalIsLoading,
-    error: bridgeProposalError,
-  } = useEasyWrite({
-    address: l1.metadataBridge,
-    chainId: l1.chain.id,
-    abi: parseAbi([
-      'function bridgeProposalMetadata(uint256 proposalId) public payable returns (uint256 sequence)',
-    ]),
-    functionName: 'bridgeProposalMetadata',
-    args: [id],
-    value: fees?.l1 || BigInt(0),
+
+  const totalVotesL1 = votes.l1.forVotes + votes.l1.againstVotes + votes.l1.abstainVotes;
+  const totalVotesNotBridged = votes.l2NotBridged
+    ? votes.l2NotBridged.forVotes +
+      votes.l2NotBridged.againstVotes +
+      votes.l2NotBridged.abstainVotes
+    : BigInt(0);
+  const totalVotes = votes.l2NotBridged ? totalVotesL1 + totalVotesNotBridged : totalVotesL1;
+
+  const makeL1VoteBar = ({
+    totalVotes,
+    votes,
+    bridged,
+    color,
+  }: {
+    totalVotes: bigint;
+    votes: bigint;
+    bridged: bigint;
+    color: string;
+  }) => {
+    // for tailwind tree-shaking...
+    const colors =
+      'text-green-500 text-green-300 text-red-500 text-red-300 text-gray-500 text-gray-300 bg-green-500 bg-green-300 bg-red-500 bg-red-300 bg-gray-500 bg-gray-300';
+    return (
+      <>
+        <div className={'flex w-full items-center text-' + color + '-500'}>
+          <span className="font-bold text-xs">
+            {nFormatter(+formatUnits(votes, l2.tokenDecimals))}
+          </span>
+          <span className={'ml-1 text-xs ' + 'text-' + color + '-300'}>
+            {bridged > BigInt(0)
+              ? `(${nFormatter(+formatUnits(bridged, l2.tokenDecimals))} bridged from L2)`
```

Comment:
Should the tokenDecimals be from L2 or from L1?

Reply:
thanks for the catch